src/rust/engine/process_execution/src/nailgun/nailgun_pool.rs

```diff
@@ -262,64 +300,113 @@ pub struct NailgunProcess {
   pub fingerprint: NailgunProcessFingerprint,
   pub build_id: String,
   pub port: Port,
-  pub handle: Arc<Mutex<std::process::Child>>,
+  pub handle: Arc<parking_lot::Mutex<std::process::Child>>,
 }
 
-fn read_port(child: &mut std::process::Child) -> Result<Port, String> {
-  let stdout = child
-    .stdout
-    .as_mut()
-    .ok_or_else(|| "No Stdout found!".to_string());
-  stdout.and_then(|stdout| {
-    let reader = io::BufReader::new(stdout);
-    let line = reader
-      .lines()
-      .next()
-      .ok_or("There is no line ready in the child's output")?
-      .map_err(|err| format!("{}", err))?;
-    let port = &NAILGUN_PORT_REGEX
-      .captures_iter(&line)
-      .next()
-      .ok_or("Output for nailgun server didn't match the regex!")?[1];
-    port
-      .parse::<Port>()
-      .map_err(|e| format!("Error parsing port {}! {}", &port, e))
+fn read_port(stdout: PathBuf) -> BoxFuture<Port, String> {
+  future::loop_fn(MAX_PROCESS_OUTPUT_POLLS, move |mut loops| {
+    let stdout = stdout.clone();
+    let wait_period = Duration::from_millis(100);
```

Comment:
100ms seems like a really long polling interval - maybe drop to 1 or 10? Particularly as presumably the `BufReader::take` has a little waiting involved? Maybe do the 100ms wait _before_ the loop, but then tight-loop or 1ms loop inside?

Reply:
Sounds good.